### DISCUSSION TITLE
Removal of initial volumes

### DISCUSSION THREAD

**Discussion by jb21469** [HALFpipe/HALFpipe#723] [Q&A] [answered]
Hi all,

I have preprocessed my resting state image data via ENIGMA HALFpipe singularity v.1.2.2. In the setup, I specified for 10 initial volumes to be removed. In the reportvals.txt file, dummy scans are correctly coded as 10 per subject. However, the output preprocessed file and correlation matrix retains the original number of volumes. I would have expected that the final files to have 10 less TRs. Does anyone have insight on this?

Thanks,
Jenna

**Comment by HippocampusGirl** [comment]
Hi @jb21469,

That seems like a bug. I will have a look and get back to you :-)

**Comment by jb21469** [comment]
Hi Lea,

Ok, thanks for looking into it :)

Jenna

**Comment by jb21469** [comment]
Hi Lea,

Though denoising was completed with the 10 initial volumes (issue stated above), is it viable to remove the 10 initial volumes from the resulting time series matrix and continue with the analysis? Or is it best practice to remove these volumes prior to the denoising regression?

Best,
Jenna

**Reply by jb21469** [reply]
Hi Lea,

Just checking back in on this. As a reminder, I specified 10 initial volumes to be removed in the setup. The 10 initial volumes are correctly tagged as non-steady state volumes in the fmriprep confounds time series file. However, all volumes are retained in the final dataset. 

To move forward with these data, is it possible to remove the 10 volumes following denoising (e.g., remove volumes from the output preprocessed files and time series matrices)? Or should the volumes be deleted before initiating ENIGMA and the trimmed data rerun through the ENIGMA pipeline?

Thank you for your advice!
Jenna

**Reply by HippocampusGirl** [reply, marked as answer]
Hi @jb21469,
I'm sorry for the late response. The issue is fixed in the `latest` version (see https://github.com/HALFpipe/HALFpipe/commit/8244ec259a91df3bef1a2e184d0f3af11fff163b), so that the initial volume removal should work as expected.

If you cannot move to the newer version, then I would advice to delete these volumes before running HALFpipe. Do you agree @ilyaveer?

Best,
Lea

**Reply by jb21469** [reply]
Hi Lea,

No worries! I was able to download and run the latest version, which successfully removed the initial volumes. Thank you for your help with this!

Best,
Jenna
